include/deal.II/base/synchronous_iterator.h

```diff
@@ -67,6 +67,16 @@ struct SynchronousIterators
   Iterators &
   operator*();
 
+  /**
+   * The following traits are necessary for std::distance() to
+   * work with SynchronousIterators.
+   */
+  using difference_type   = std::size_t;
+  using iterator_category = std::bidirectional_iterator_tag;
```

Comment:
This is probably safe, though it really depends on the type of the underlying iterator types. Let's keep it this way for the moment, we can revisit this later.